src/linter.rs

```diff
@@ -192,11 +192,15 @@ impl Linter {
         top_level_ctxt,
       );
 
+      let mut sorted_rules = self.rules.clone();
+      crate::rules::sort_rules_by_priority(&mut sorted_rules);
```

Comment:
Is this clone necessary? It seems to me that `crate::rules::sort_rules_by_priority(&mut self.rules)` will just work fine.

Reply:
Good catch